Yoast/javascript
# Add multiletter prefixes for Arabic

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds prefixes `وبال, مست, تنن, است, نست, ولل, كال, فال, بال, وال, لن, سن, ست, ات, وس, فل, فب, فس, لل, ول, وب, ال`  to the list of ignored prefixes for Arabic. When the Arabic keyphrase is preceded by one of these prefixes, the basic keyphrase forms will also be created.
* [Yoast SEO Free] Makes it also possible to recognize keyphrases in Arabic when they are preceded by one of these prefixes, e.g. `والقبعة`, `فسحرية`, or `وبالمفتاح`.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test` and make sure all tests pass.

* Test in the content-analysis app:
        * Start the content-analysis app by running `yarn start`
        * Paste a text of at least 150 words
        * Set the locale to `ar` 
        * Switch the morphology button to Off
        * Set the keyphrase as `مفتاح ` and paste `وبالمفتاح` in the text 
        * Check in the assessment that the keyword is found in the text
        * Check that the functionality still works when the keyphrase and form in the text are switched around
        * Check that the keyphrase `مفتاح` matches also a form with a different prefix (e.g., `لنمفتاح` or `كالمفتاح`)


## QA Test instructions 
This PR can be tested by following these steps:
* Test the following functionality in Free:

* Set your site to Arabic 
* As your keyphrase, set the word `طائر`.
* Add a text with at least 150 words.
* In the text, add the form `وبالطائر`. The keyphrase density assessment should show 1 keyphrase occurrence. When clicking on the marker for the assessment, `وبالطائر` should be marked.
* Switch around the keyphrase and the word form.
* The keyphrase density assessment should still show 1 keyphrase occurrence.
* Switch the form in the text to a form with another prefix, e.g. `كالطائر` or `وللطائر`.
The keyphrase density assessment should still show 1 keyphrase occurrence.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-204
